webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/Hdf5FileCache.scala

```diff
@@ -41,3 +48,26 @@ class Hdf5FileCache(val maxEntries: Int) extends LRUConcurrentCache[String, Cach
     }
   }
 }
+
+object CachedHdf5Utils {
+  def safeExecute[T](filePath: Path, meshFileCache: Hdf5FileCache)(block: CachedHdf5File => T)(
+      implicit ec: ExecutionContext): Fox[T] =
+    for {
+      _ <- bool2Fox(filePath.toFile.exists()) ?~> "mesh.file.open.failed"
+      result <- Using(meshFileCache.withCache(filePath)(CachedHdf5File.fromPath)) {
+        block
+      }.toFox
+    } yield result
+
+  def safeExecuteBox[T](filePath: Path, meshFileCache: Hdf5FileCache)(block: CachedHdf5File => T): Box[T] =
+    for {
+      _ <- if (filePath.toFile.exists()) {
+        Full(true)
+      } else {
+        Empty ~> "mesh.file.open.failed"
```

Comment:
```suggestion
        Failure("mesh.file.open.failed")
```
(I believe the ~> operator turns the Empty box into a Failure anyway)